SlicerLite/SlicerLiteLib/SlicerLiteModuleWidget.py

```diff
@@ -0,0 +1,170 @@
+
+        # Need to add a new slot in order to avoid 3DSlicer to update this button visibility when adding a new segment
+        if add_segmentation_button:
+            add_segmentation_button.clicked.connect(lambda _: self.segment_editor_widget.rotateSliceViewsToSegmentation())
+
+    def on_click_load_dicom_directory(self) -> None:
+        """
+        User choose directory where DICOM will be extracted
+        """
+        dirPath = qt.QFileDialog.getExistingDirectory(self,
+                                                     "Choose dicom file directory to load",
+                                                     self._lastOpenedDirectory)
+        if not dirPath:
+            return
+        self.load_dicom_directory(dirPath)
+
+    def load_dicom_directory(self, directory_path: str) -> None:
+        """
+        Add and load the input dicom dir into the DICOM database
+        directory_path: Path to the directory that contains dicoms
+        """
+        qt.QApplication.setOverrideCursor(qt.Qt.WaitCursor)
+
+        self._lastOpenedDirectory = directory_path
+        loadedVolumesNodes = self.data_loader.load_dicom_dir_in_db_and_extract_volumes_as_items(self._lastOpenedDirectory)
+        for volumeNode in loadedVolumesNodes:
+            self.item_table_model.add_item(Item(volumeNode))
+
+        qt.QApplication.restoreOverrideCursor()
+
+        # Don't show anything in slices
+        SlicerUtils.show_volume_as_foreground_in_slices(None)
+        SlicerUtils.show_volume_as_background_in_slices(None)
+        self.item_table_view.clearSelection()
+
+    def set_current_item(self, item: Item) -> None:
+        """
+        Set the current item to the corresponding modules
+        """
+        self.rendering_module.setMRMLVolumeNode(item.volumeNode if item else None)
+        self.shift_slider_widget.setEnabled(bool(item.volumeNode) if item else False)
+        self.segment_editor_widget.setSegmentationNode(item.segmentationNode if item else None)
+        self.segment_editor_widget.setSourceVolumeNode(item.volumeNode if item else None)
+        if item:
+            self.segment_editor_widget.rotateSliceViewsToSegmentation()
+
+    def on_table_view_item_clicked(self, modelIndex: qt.QModelIndex) -> None:
+        """
+        Update views and current item according to the clicked item
+        """
+        if not modelIndex.model():
+            SlicerUtils.show_volume_as_foreground_in_slices(None)
+            SlicerUtils.show_volume_as_background_in_slices(None)
+            self.item_table_view.clearSelection()
+            self.set_current_item(None)
+            return
+
+        item = modelIndex.model().item(modelIndex.row()).data(ItemModel.ItemUserRole)
+        # We only want to select line if user click on volume's name
+        if modelIndex.column() != 0:
+            return
+
+        self.set_current_item(item)
+        for button_delegate in self.buttons_delegate:
```

Comment:
Breaks encapsulation. The table shouldn't have to tell its delegate the selection row.

Instead you should probably use the openPersistentEditor for the new row and close the persistent editor for the previous one.